src/interface/TWFilecoinSigner.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <TrustWalletCore/TWFilecoinSigner.h>
+
+#include "../Filecoin/Signer.h"
+#include "../Filecoin/Transaction.h"
+#include "../PrivateKey.h"
+#include "../proto/Filecoin.pb.h"
+
+using namespace TW;
+using namespace TW::Filecoin;
+
+TW_Filecoin_Proto_SigningOutput TWFilecoinSignerSign(TW_Filecoin_Proto_SigningInput data) {
+    Proto::SigningInput input;
+    input.ParseFromArray(TWDataBytes(data), static_cast<int>(TWDataSize(data)));
+
+    auto key = PrivateKey(Data(input.private_key().begin(), input.private_key().end()));
```

Comment:
we're going to simplify Signer C interface soon, could you please add method to `Filecoin::Signer`to handle `TW_Filecoin_Proto_SigningInput` directly? An example for you to reference: https://github.com/TrustWallet/wallet-core/blob/master/src/interface/TWAlgorandSigner.cpp

Reply:
I like the new pattern, I applied changes to follow exact style of Algorand:
 * Exclude signature from Transaction class
 * static Signer methods
 * Transaction.serialize(Signature &) method